packages/teleport/src/components/SelectFilter/SelectFilter.tsx

```diff
@@ -0,0 +1,328 @@
+    applyFilters(updatedFilters);
+  }
+
+  function onFilterApply() {
+    setShowSelector(false);
+    applyFilters(selectedFilters);
+  }
+
+  function handleKeyDown(e: KeyboardEvent) {
+    if (e.key === 'Escape') {
+      setShowSelector(false);
+    }
+  }
+
+  React.useEffect(() => {
+    setSelectedFilters(appliedFilters);
+  }, [appliedFilters]);
+
+  React.useEffect(() => {
+    function handleOnClick(e) {
+      // Ignore event for clicking near buttons.
+      if (e.target.closest('button')) {
+        return;
+      }
+
+      // If event is not from inside the select wrapper, close the selector.
+      if (!selectWrapperRef.current?.contains(e.target)) {
+        setShowSelector(false);
+      }
+    }
+
+    window.addEventListener('click', handleOnClick);
+    return () => window.removeEventListener('click', handleOnClick);
+  }, []);
+
+  const $labels = appliedFilters.map((o, key) => (
+    <Label key={key} name={o.label} onClick={() => deleteLabel(o.label)} />
+  ));
+
+  return (
+    <Box mb={mb}>
+      <Flex flexWrap="wrap">
+        <Box style={{ position: 'relative' }}>
+          <AddButton
+            pl={2}
+            pr={3}
+            onClick={() => setShowSelector(!showSelector)}
+            mt={0}
+            mr={3}
+            mb={2}
+          >
+            <Add fontSize={4} mr={1} color="rgba(255,255,255,0.24)" />
+            Add Filters
+          </AddButton>
+          {showSelector && (
+            <Box
+              mt={-2}
+              bg="#fff"
+              borderRadius={2}
+              borderTopLeftRadius={0}
+              style={{ position: 'absolute', zIndex: 1 }}
+            >
+              <Flex
+                justifyContent="space-between"
+                alignItems="center"
+                px={3}
+                pb={0}
+              ></Flex>
+              <StyledSelect ref={selectWrapperRef}>
+                <Select
+                  autoFocus
+                  placeholder="Search..."
+                  value={selectedFilters}
+                  options={filters}
+                  isSearchable={true}
+                  isClearable={false}
+                  isMulti={true}
+                  menuIsOpen={true}
+                  hideSelectedOptions={false}
+                  controlShouldRenderValue={false}
+                  onChange={(o: Option[]) => setSelectedFilters(o)}
+                  onKeyDown={handleKeyDown}
+                  components={{
+                    Option: OptionComponent,
+                    Control: ControlComponent,
+                  }}
+                  customProps={{
+                    onFilterApply,
+                    appliedFilters,
+                    selectedFilters,
+                    clearFilters,
+                  }}
+                />
+              </StyledSelect>
+            </Box>
+          )}
+        </Box>
+        {$labels}
+      </Flex>
+    </Box>
+  );
+}
+
+const ControlComponent = props => {
```

Comment:
It would be good to have `ControlComponent`, `OptionComponent` and `Label` typed

Reply:
i typed `Label`, but for the others, tsc blew up with type errors after I installed `@types/react-select`, so i created an issue to handle refactoring later: https://github.com/gravitational/webapps/issues/460